docs/nia: update for 0.6 GA (#13191)
hashicorp/consul: website/content/docs/release-notes/consul-terraform-sync/v0_6_x.mdx

---
layout: docs

> page_title: 0.4.x

Reviewer: Should be 0.6.x

Author: Ah, thanks for catching

Author: ```suggestion
page_title: 0.6.x
```

description: >-
  Consul-Terraform-Sync release notes for version 0.6.x
---

# Consul-Terraform-Sync 0.6.0

## Release Highlights

> Per-Task Execution Mode with Support for Terraform Cloud Agent (Enterprise): CTS now allows users to select the Terraform execution mode to be “remote” or “cloud agent” per Task. For more information refer to the [per-task execution mode documentation](/docs/nia/network-drivers/terraform-cloud#remote-workspaces).

Reviewer: Should we use the Enterprise tag instead?

Reviewer: ```suggestion
Per-Task Execution Mode with Support for Terraform Cloud Agent <EnterpriseAlert inline />: CTS now allows users to select the Terraform execution mode to be “remote” or “cloud agent” per Task. For more information refer to the [per-task execution mode documentation](/docs/nia/network-drivers/terraform-cloud#remote-workspaces).
```

> HCP Consul support (Enterprise): CTS now supports interoperability with HCP Consul with license retrieval capabilities that allow users to keep licenses in sync with their HCP Consul or Consul enterprise deployment.

Reviewer: ```suggestion
HCP Consul support <EnterpriseAlert inline />: CTS now supports interoperability with HCP Consul with license retrieval capabilities that allow users to keep licenses in sync with their HCP Consul or Consul enterprise deployment.
```

> Auto Service-Registration with Consul, and Health API Endpoint: CTS now supports a new configuration block that auto registers CTS as a service within Consul as soon as the CTS is instantiated. This eliminates the manual step of registering CTS as a service within the Consul catalog. This service registration also enables health checking on CTS using the newly added health API endpoint.

Reviewer: ```suggestion
Auto Service-Registration with Consul, and Health API Endpoint: CTS now supports a new configuration block that auto registers CTS as a service within Consul as soon as CTS is instantiated. This eliminates the manual step of registering CTS as a service within the Consul catalog. This service registration also enables health checking on CTS using the newly added health API endpoint.
```

Reviewer: Should we add a link to the health API endpoint docs?

Author: Good idea, I updated your suggestion to include the health endpoint url

CLI discoverability: CTS now improves discoverability in the CLI. The `consul-terraform-sync` command features opt-in autocompletion for flags, subcommands, and arguments

Health Check API: CTS introduces a new [API endpoint](/docs/nia/api/health#health) that provides operators with the health status of their CTS instances.

> ## Supported Software Versions

Reviewer: Should we mention HCP-Consul?

Author: @devarshishah3 - could I get your input on this?

Reviewer 2: Good catch, Michael. Yes

(Enterprise) HCP Consul - Latest

> Per-Task Execution Mode with Support for Terraform Cloud Agent <EnterpriseAlert inline />: CTS now allows users to select the Terraform execution mode to be “remote” or “cloud agent” per Task. For more information refer to the [per-task execution mode documentation](/docs/nia/network-drivers/terraform-cloud#remote-workspaces).
> 
> HCP Consul support <EnterpriseAlert inline />: CTS now supports interoperability with HCP Consul with license retrieval capabilities that allow users to keep licenses in sync with their HCP Consul or Consul enterprise deployment.
> 
> Auto Service-Registration with Consul, and Health API Endpoint: CTS now supports a new configuration block that auto registers CTS as a service within Consul as soon as CTS is instantiated. This eliminates the manual step of registering CTS as a service within the Consul catalog. This service registration also enables health checking on CTS using the newly added [health API endpoint](/docs/nia/api/health#health).
> 
> CLI discoverability: CTS now improves discoverability in the CLI. The `consul-terraform-sync` command features opt-in autocompletion for flags, subcommands, and arguments
> 
> Health Check API: CTS introduces a new [API endpoint](/docs/nia/api/health#health) that provides operators with the health status of their CTS instances.

Reviewer 3: ```suggestion

We have implemented the following features in this release:
 
### Per-task Execution Mode with Support for Terraform Cloud Agent <EnterpriseAlert inline /> 

You can now execute Terraform tasks in `remote` or `cloud agent` mode. For more information, refer to the [per-task execution mode documentation](/docs/nia/network-drivers/terraform-cloud#remote-workspaces).

### HCP Consul Cupport <EnterpriseAlert inline /> 

CTS now supports interoperability with HCP Consul. CTS retrieves licenses from HCP Consul so that users can keep their HCP Consul license or Consul enterprise deployment license in sync.

### Auto Service-registration with Consul and Health API Endpoint 

CTS includes a new configuration block that automatically registers CTS as a service within Consul as soon as CTS is instantiated. This eliminates the manual step of registering CTS as a service within the Consul catalog. This service registration also enables health checking on CTS using the newly added [health API endpoint](/docs/nia/api/health#health).

### CLI Discoverability 

We improved CTS's CLI discoverability functionality. The `consul-terraform-sync` command features opt-in autocompletion for flags, subcommands, and arguments

### Health Check API 

CTS includes a new [API endpoint](/docs/nia/api/health#health) that provides operators with the health status of their CTS instances.
```
The formatting was weird with title style sentences, so I added an intro sentence below section heading and turned each label-type statement into its own heading. Alternatively, you could change the subheads into a bulleted list. I though that the subhead format might be better because a few descriptions are paragraph length. Also, it would be easier to link to a specific point.

Author: Thanks for the suggestions @trujillo-adam - they look good. Committed!

- Consul: v1.9+

> - Terraform CLI: v0.13. - v1.1

Reviewer: ```suggestion
- Terraform CLI: v0.13 - v1.1
```

> - Terraform Cloud: Latest

Reviewer 2: Can we prepend "(Enterprise)" in front Terraform Cloud: Latest and Terraform Enterprise: v202010-2- Latest

Reviewer: Suggest using the alert tag instead: <EnterpriseAlert inline />

Reviewer: Suggest using the alert tag instead: `<EnterpriseAlert inline />`

```suggestion
- <EnterpriseAlert inline /> Terraform Cloud: Latest
```

Author: Didn't refresh and missed this suggestion, sorry Michael

> - Terraform Cloud: Latest

Author: ```suggestion
- (Enterprise) Terraform Cloud: Latest
```

> - Terraform Enterprise: v202010-2- Latest

Reviewer: ```suggestion
- Terraform Enterprise: v202010-2 - Latest
```

Author: Thanks for all these fixes! FYI I updated your suggestion with Devarshi's suggestion prepend Enterprise

> - Terraform Enterprise: v202010-2- Latest

Author: ```suggestion
- (Enterprise) Terraform Enterprise: v202010-2- Latest
```

## What's Changed

**Deprecations**

> * **(Enterprise Only)** Deprecate `terraform_version` task configuration in favor of `terraform_cloud_workspace.terraform_version` task configuration [[GH-790](https://github.com/hashicorp/consul-terraform-sync/issues/790)]

Reviewer: ```suggestion
<EnterpriseAlert inline /> Deprecate `terraform_version` task configuration in favor of `terraform_cloud_workspace.terraform_version` task configuration [[GH-790](https://github.com/hashicorp/consul-terraform-sync/issues/790)]
```

> * **(Enterprise Only)** Deprecate `license_path` configuration in favor of `license.path` configuration [[GH-810](https://github.com/hashicorp/consul-terraform-sync/issues/810)]

Reviewer: ```suggestion
<EnterpriseAlert inline /> Deprecate `license_path` configuration in favor of `license.path` configuration [[GH-810](https://github.com/hashicorp/consul-terraform-sync/issues/810)]
```

* Deprecate starting CTS as a daemon using no CLI command in favor of starting CTS as a daemon using the new `start` command [[GH-866](https://github.com/hashicorp/consul-terraform-sync/issues/866)]

## Changelogs

The changelogs for this major release version and any maintenance versions are listed below.

> -> **Note**: These links will take you to the changelogs on the GitHub website.

Reviewer: ```suggestion
-> **Note**: These links will redirect you to the changelogs on the GitHub website.
```

---
layout: docs
page_title: 0.6.x
description: >-
  Consul-Terraform-Sync release notes for version 0.6.x
---

# Consul-Terraform-Sync 0.6.0

## Release Highlights

We have implemented the following features in this release:
 
### Per-task Execution Mode with Support for Terraform Cloud Agent <EnterpriseAlert inline /> 

You can now execute Terraform tasks in `remote` or `cloud agent` mode. For more information, refer to the [per-task execution mode documentation](/docs/nia/network-drivers/terraform-cloud#remote-workspaces).

> ### HCP Consul Cupport <EnterpriseAlert inline /> 

Author: Thanks for catching all of these!

CTS now supports interoperability with HCP Consul. CTS retrieves licenses from HCP Consul so that users can keep their HCP Consul license or Consul enterprise deployment license in sync.

### Auto Service-registration with Consul and Health API Endpoint 

CTS includes a new configuration block that automatically registers CTS as a service within Consul as soon as CTS is instantiated. This eliminates the manual step of registering CTS as a service within the Consul catalog. This service registration also enables health checking on CTS using the newly added [health API endpoint](/docs/nia/api/health#health).

### CLI Discoverability 

> We improved CTS's CLI discoverability functionality. The `consul-terraform-sync` command features opt-in autocompletion for flags, subcommands, and arguments

Reviewer: Maybe link to the relevant autocomplete docs?

### Health Check API 

> CTS includes a new [API endpoint](/docs/nia/api/health#health) that provides operators with the health status of their CTS instances.

Reviewer: mix of singluar status and plural instances? Maybe:

```suggestion
CTS includes a new [API endpoint](/docs/nia/api/health#health) that provides operators with the health statuses of their CTS instances.
```
